packages/perseus/src/types.ts

```diff
@@ -44,6 +49,9 @@ export type SerializedState = Record<string, any>;
  * scoring, state serialization/deserialization, and focus management.
  */
 export interface Widget {
+    // don't use getType; it's just a dummy method to help TypeScript's weak
+    // typing to recognize non-interactive widgets as Widgets
+    getType?: () => string;
```

Comment:
I expect push-back on this one. Here's the problem: when I removed all the validation methods on non-interactive widgets I got TS errors to the effect of "there's no overlap between Passage and Widget." That's because non-interactive widgets don't implement _any_ of the optional methods on Widget and Widget only has optional methods.

By adding this silly method and having non-interactive widgets implement it, TS suddenly recognizes them as Widgets.

What I didn't want them to do is implement _real_ methods they didn't need to because I think that's confusing long-term. For instance if we made them all return `null` for `getUserInput`, suddenly we're having to maintain code that expects `null` as a possibility.

I wanted to make this a required method, but it broke:

```TS
type WidgetExports<T extends React.ComponentType<any> & Widget = React.ComponentType<any>>
```

I don't know why.

Reply:
I originally tried a required method, but this blew up and I didn't want to go down a rabbit hole:

``` TS
export type WidgetExports<
    T extends React.ComponentType<any> & Widget = React.ComponentType<any>,
>
```
But I moved the widgets that needed it from "getType" to "isWidget".

> I'm not sure I agree. getUserInput is currently typed as...

This was changed in this PR to be `() => UserInputArray | UserInput | undefined` and the only reason `undefined` is there is because of Group (and I'd like to remove that).